packages/android_alarm_manager/lib/android_alarm_manager.dart

```diff
@@ -54,6 +54,31 @@ void _alarmManagerCallbackDispatcher() {
   _channel.invokeMethod<void>('AlarmService.initialized');
 }
 
+/// This is a wrapper around a few static functions so that unit tests can mock
+/// them out in the test environment. This is only visible for the unit tests and
+/// should not be accessed directly by users of the plugin.
+@visibleForTesting
+class InstanceWrappers {
```

Comment:
Will this appear in the docs for the plugin?

Reply:
Yes. Technically this was a "missing doc" before since it's public and visible to users of the plugin. I figured a warning saying not to touch this made the most sense as the only user-facing documentation on it.

Reply:
I'm really not a fan of polluting the API surface with code only used for testing. If this wasn't going to appear in the docs that would be a different story, but it would be great if we could hide this somewhere.

Reply:
Never mind, I'm going to delete this anyway as @cyanglaz suggested in https://github.com/flutter/plugins/pull/2283#discussion_r349196050.

Reply:
Sorry, Github hadn't loaded the earlier comment at the time I posted my last one.

Ideally I'd like to avoid all `@visibleForTesting` identifiers too but I'm not sure how to completely do it here. `DateTime.now` and `PluginUtilities.getCallbackHandle` are both static and referenced statically internally. `AndroidAlarmManager` itself is static so I can't use constructor injection. The unit tests can only touch the class through its public API surfaces. Right now I am cutting down on the visibility a little bit from what's currently in this patch, but I'm still adding this method to `AndroidAlarmManager` itself:

```dart
  /// This is exposed for the unit tests. It should not be accessed by users of
  /// the plugin.
  @visibleForTesting
  static void setTestOverides({Now now, GetCallbackHandle getCallbackHandle}) {
    _now = now;
    _getCallbackHandle = getCallbackHandle;
  }
```

And then the class internally is using `_now` and `_getCallbackHandle` instead of the static functions.

If you have any ideas on how to avoid it totally I would be happy to hear it, I'd love to avoid this too. I'm also exposing the MethodChannel so I can test it too.